sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -394,9 +420,20 @@ public virtual async Task<Response> UploadAsync<T>(string ruleId, string streamN
                 else
                 {
                     Exception exceptionEventHandler = await ProcessCompletedTaskEventHandlerAsync(task.CurrentTask, task.Logs, options, cancellationToken).ConfigureAwait(false);
-                    shouldAbort = exceptionEventHandler != null;
-                    if (shouldAbort)
+                    logsFailed += task.Logs.Count;
```

Comment:
This doesn't seem right - it looks like we call ProcessCompletedTask even for successful responses, but we are still adding the logs to the failed count.